Products/zms/zmslog.py

```diff
@@ -1,217 +1,221 @@
+
+
+################################################################################
+###
+###   Class
+###
+################################################################################
+class ZMSLog(ZMSItem.ZMSItem):
+
+    # Properties.
+    # -----------
+    meta_type = 'ZMSLog'
+    zmi_icon = "fas fa-bug"
+    icon_clazz = zmi_icon
+
+
+    # Management Options.
+    # -------------------
+    def manage_options(self):
+      return (
+        { 'label': 'Settings','action': '../manage_customize'},
+        )
+
+    # Management Interface.
+    # ---------------------
+    manage_main = PageTemplateFile( 'zpt/ZMSLog/manage_main', globals())
+    manage_remote = PageTemplateFile( 'zpt/ZMSLog/manage_remote', globals())
+
+    LOGGER = logging.getLogger("ZMS")
+
+    ############################################################################
+    #  ZMSLog.__init__: 
+    #
+    #  Initialise a new instance of ZMSLog.
+    ############################################################################
+    def __init__(self, copy_to_stdout=False, logged_entries=[ 'ERROR']):
+      self.id = 'zms_log'
+      self.copy_to_stdout = copy_to_stdout
+      self.logged_entries = logged_entries
+
+    ############################################################################
+    #  ZMSLog.setProperties: 
+    #
+    #  Set properties.
+    ############################################################################
+    def setProperties(self, REQUEST, RESPONSE): 
+      """ ZMSLog.setProperties """
+      self.tail_event_log_linesback = REQUEST.get('tail_event_log_linesback', 100)
+      self.copy_to_stdout = 'copy_to_stdout' in REQUEST
+      self.logged_entries = REQUEST.get( 'logged_entries', [])
+      return RESPONSE.redirect( REQUEST[ 'HTTP_REFERER'])
+
+    # --------------------------------------------------------------------------
+    #  ZMSLog.hasSeverity:
+    # --------------------------------------------------------------------------
+    def hasSeverity(self, severity):
+      return severity_string(severity) in self.logged_entries
+
+    # --------------------------------------------------------------------------
+    #  ZMSLog.LOG:
+    # --------------------------------------------------------------------------
+    def LOG(self, severity, info):
+      log_severity = severity
+      if log_severity == logging.DEBUG:
+        log_severity = logging.INFO
+      self.LOGGER.log( severity, info)
+      if getattr( self, 'copy_to_stdout', True):
+        standard.writeStdout(self, '%s %s(%i) %s'%(str(log_time()), severity_string(severity), int(severity), info))
+
+    # --------------------------------------------------------------------------
+    #  ZMSLog.getLOG:
+    # --------------------------------------------------------------------------
+    def getLOG(self, REQUEST, RESPONSE=None):
+      """ ZMSLog.getLOG """
+      filename = os.path.join(standard.getINSTANCE_HOME(), 'log', 'zms-event.log')
```

Comment:
check Zope API to get filepath from zope.ini

Reply:
DONE by https://github.com/zms-publishing/ZMS/commit/8dc3dabafdffe14a45352b8e9680fe9cfc666676